common/src/path.rs

```diff
@@ -65,25 +110,19 @@ pub fn join_path(base: &str, other: &str) -> String {
         }
     }
 
-    let win_style = is_windows_path(base) || is_windows_path(other);
-
-    if win_style {
-        format!(
-            "{}\\{}",
-            base.trim_end_matches(&['\\', '/'][..]),
-            other.trim_start_matches(&['\\', '/'][..])
-        )
-    } else {
-        format!(
-            "{}/{}",
-            base.trim_end_matches('/'),
-            other.trim_start_matches('/')
-        )
-    }
+    // Always trim by both separators, since as soon as the path is Windows, slashes also count as
+    // valid path separators. However, use the main separator for joining.
+    let is_windows = is_windows_path(base) || is_windows_path(other);
+    format!(
+        "{}{}{}",
+        base.trim_end_matches(is_path_separator),
+        if is_windows { '\\' } else { '/' },
+        other.trim_end_matches(is_path_separator)
```

Comment:
```suggestion
        other.trim_start_matches(is_path_separator)
```

Also please add a test that covers this.

Reply:
Good. Catch. I'm glad the tests didn't catch this.

Reply:
I think this doesn't even make sense.